internal/cmds/setting/argument/strings.go

```diff
@@ -0,0 +1,72 @@
+package argument
+
+import (
+	"encoding/csv"
+	goerrors "errors" // TODO: remove
+	"reflect"
+	"strings"
+)
+
+// TODO: better name?
+// TODO: support Stringer?
+type StringsConstraint interface{ string | []string }
```

Comment:
>better name?

Maybe just `Strings`? `StringArgs`? `ArgStrings`? `ArgumentStrings`?
Any of these seem better than the placeholder.
... Actually, since this is only ever used in `ParseStrings` we can just inline it.
```go
func ParseStrings[stringish string | []string](arg Argument, value stringish,
	parsers ...TypeParser,
) (any, error) {
```

I also think we can eliminate the need for this within the `request` pkg which would allow us to unexport this and move it within `environment` for now.

That would require rethinking how `cmds.Option`s are generated.
Instead of using their `StringOption` for non-builtin types and parsing it ourselves, we can probably just implement `cmds.Option` with our own generic wrapper for non-builtin types and let the cmdslib handle parsing them into its `request.Options` map.
Then we'd only be responsible for parsing non-CLI sources, like the environment, config files, etc.
Needs thought and will update #6 too (if it turns out to be a good idea).

>support Stringer

This would be as easy as adding it to the constraint and then checking for it in the type switch, but I don't want to do that until/unless we actually need that somewhere. (we probably won't)

Reply:
Initial version of the `cmds.Option` constructor was done in: 3bed275d7b10007348ae473e7f82ef7bc2e44a49
After the argument side of this is investigated, that commit should get cherry-picked into the other branch, and we'll rebase off it.

This removes the need for us to parse `string` values ourselves, instead letting the cmdslib handle it internally (with our provided parsers).

Unfortunately, the cmdslib has special cases for `cmds.StringsOption`, which might cause some grief. (ref: cmds/cli/parse.go `setOpts`)
It basically allows multiple values to be provided in a variety of ways, where other option types have more parsing restrictions (like only allowing flags once with 1 value)

I'll have to consider how best to handle this.
The upstream interface is kind of confusing. `Option.Type` doesn't actually return a `Type`, but a `Kind`. And despite the documentation, seems to pertain more to the option itself, rather than its value.
Worse still, the `Kind` for a `StringOption` isn't `reflect.Slice`, but `reflect.Array`, despite the fact the value returned is `[]string`...

I'm gonna see if hacking around this causes any problems. If our kind is a slice, just lie and say it's an array from our `Type` method (like upstream does).
This should be good enough until we have time to replace that library altogether.

Reply:
The hack mentioned above won't work unfortunately.
Testing as-is we get this `options_test.go:325: multiple values for option "test"` when trying to use a slice type for our argument values.
The hack above will bypass this, but internally the cmdslib does type assertions everywhere, which see right through our charade. `panic: interface conversion: interface {} is []time.Duration, not []string`

So as far as I can tell we're stuck with using cmds.StringsOption and doing a second round of parsing ourselves if our values are vectors.
But this still probably lets us get away with changing `ParseStrings` so that it has a more sensible signature, like 
`ParseStrings(strings []string, optionalUserParsers ...TypeParser)`